packages/datetime/src/components/shortcuts/shortcuts.tsx

```diff
@@ -75,51 +75,47 @@ export interface DatePickerShortcutMenuProps {
  * This component may be used for single date pickers as well as range pickers by toggling the
  * `useSingleDateShortcuts` option.
  */
-export class DatePickerShortcutMenu extends React.PureComponent<DatePickerShortcutMenuProps> {
-    public static defaultProps: Partial<DatePickerShortcutMenuProps> = {
-        selectedShortcutIndex: -1,
-    };
-
-    public render() {
-        const shortcuts =
-            this.props.shortcuts === true
-                ? createDefaultShortcuts(
-                      this.props.allowSingleDayRange,
-                      this.props.timePrecision !== undefined,
-                      this.props.useSingleDateShortcuts === true,
-                  )
-                : this.props.shortcuts;
-
-        const shortcutElements = shortcuts.map((shortcut, index) => (
-            <MenuItem
-                active={this.props.selectedShortcutIndex === index}
-                disabled={!this.isShortcutInRange(shortcut.dateRange)}
-                key={index}
-                onClick={this.getShorcutClickHandler(shortcut, index)}
-                shouldDismissPopover={false}
-                text={shortcut.label}
-            />
-        ));
-
-        return (
-            <Menu aria-label="Date picker shortcuts" className={Classes.DATERANGEPICKER_SHORTCUTS} tabIndex={0}>
-                {shortcutElements}
-            </Menu>
-        );
-    }
-
-    private getShorcutClickHandler = (shortcut: DateRangeShortcut, index: number) => () => {
-        const { onShortcutClick } = this.props;
-
-        onShortcutClick(shortcut, index);
-    };
-
-    private isShortcutInRange = (shortcutDateRange: DateRange) => {
-        const { minDate, maxDate } = this.props;
-
-        return isDayRangeInRange(shortcutDateRange, [minDate, maxDate]);
-    };
-}
+export const DatePickerShortcutMenu: React.FC<DatePickerShortcutMenuProps> = props => {
+    const {
+        allowSingleDayRange = false,
+        maxDate,
+        minDate,
+        onShortcutClick,
+        selectedShortcutIndex = -1,
+        timePrecision,
+        useSingleDateShortcuts = false,
+    } = props;
+
+    const shortcuts =
+        props.shortcuts === true
+            ? createDefaultShortcuts(allowSingleDayRange, timePrecision !== undefined, useSingleDateShortcuts)
+            : props.shortcuts;
+
+    const isShortcutInRange = React.useCallback(
+        (shortcutDateRange: DateRange) => isDayRangeInRange(shortcutDateRange, [minDate, maxDate]),
+        [minDate, maxDate],
+    );
+
+    const handleShortcutClick = React.useCallback(
+        (shortcut: DateRangeShortcut, index: number) => () => onShortcutClick(shortcut, index),
+        [onShortcutClick],
+    );
```

Comment:
Curious for your thoughts here but generally don't like the get click handler callback format. I'd prefer we have a `<ShortcutMenuItem>` component that takes in `onShortcutClick` and binds a specific `useCallback` for itself.

Reply:
Yeah, I can get on board with that. Change added in cc02cf3bf2115f0d553152a926a38027b00ee053